api/models/FlaggedItem.js

```diff
@@ -1,8 +1,35 @@
+import { values } from 'lodash'
+import { validateFlaggedItem } from 'hylo-utils/validators'
+
 module.exports = bookshelf.Model.extend({
   tableName: 'flagged_items',
 
   user: function () {
     return this.belongsTo(User, 'user_id')
   }
 
+}, {
+  Category: {
+    ABUSIVE: 'abusive',
+    ILLEGAL: 'illegal',
+    OTHER: 'other',
+    SAFETY: 'safety',
+    SPAM: 'spam'
+  },
+
+  create: function (attrs) {
+    const { category, link, reason } = attrs
+
+    if (!values(this.Category).find(c => category === c)) {
+      return Promise.reject('Unknown category.')
+    }
+
+    const invalidReason = validateFlaggedItem.reason(reason)
+    if (invalidReason) Promise.reject(invalidReason)
+
+    const invalidLink = validateFlaggedItem.link(link)
+    if (invalidLink) return Promise.reject(invalidLink)
+
+    return FlaggedItem.forge(attrs).save()
```

Comment:
it's safe to use `this` here instead of `FlaggedItem`, i believe -- i know it doesn't matter much, but it feels better to use a local reference than a global one where possible

Reply:
Will do!